src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -434,15 +434,34 @@ public void EnsureSolutionIsLoaded()
             });
         }
 
-        public string SolutionDirectory => NuGetUIThreadHelper.JoinableTaskFactory.Run(GetSolutionDirectoryAsync);
+        private string _solutionDir;
+        public string SolutionDirectory
```

Comment:
The VSSolutionManager is shared across solutions.

Reply:
There's also some timing concerns https://github.com/NuGet/NuGet.Client/pull/4879 is trying to address that are not being considered here.

Reply:
What do you mean by VSSolutionManager is shared across solutions?
Could we load multiple solutions same time?

Reply:
VSSolutionManager is a shared component. Check the Importing declaration.

- open solution A
- Close solution
- Open Solution B


As long as you don't turn close VS and shut down the process, it's the same solution manager.

Reply:
I would just try to figure out what @zivkan's PR was at as he was trying to solve the same problem.

Reply:
> VSSolutionManager is a shared component. Check the Importing declaration.
> 

What is the Importing declaration?

> * open solution A
> * Close solution
> * Open Solution B
> 
> As long as you don't turn close VS and shut down the process, it's the same solution manager.

But I do reset the` _solutionDirectory` to null every time we close the solution. Isn't that enough?

Reply:
> What is the Importing declaration?

This indicates it's shared in the process: https://github.com/NuGet/NuGet.Client/blob/c0371f7165bea201aded1a8013ca0b17e2ad0092/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs#L39-L40

Reply:
Resolving this as you reset the value.